src/components/patterns/on-the-page.tsx

```diff
@@ -0,0 +1,146 @@
+"use client"
+
+import React, {useState, useEffect} from "react"
+import {Link, Maybe} from "@/lib/gql/__generated__/drupal.d"
+import {twMerge} from "tailwind-merge"
+import DrupalLink from "./elements/drupal-link"
+import SelectList from "./elements/select-list"
+import {SelectOptionDefinition} from "@mui/base/useSelect"
+import {useRouter} from "next/navigation"
+
+interface OnThePageProps {
+  relLinkHeading?: Maybe<string>
+  relLinks?: Maybe<Link[]>
+}
+
+interface Heading {
+  text: string
+  id: string
+}
+
+const OnThePageLink = ({relLinkHeading, relLinks}: OnThePageProps) => {
+  const router = useRouter()
+  const [headings, setHeadings] = useState<Heading[]>([])
+  const [activeHeading, setActiveHeading] = useState<string>("")
+
+  useEffect(() => {
+    const wysiwygContainers: NodeListOf<Element> = document.querySelectorAll(".wysiwyg")
+
+    const allHeadings: Heading[] = []
+
+    wysiwygContainers.forEach((container, containerIndex) => {
+      const h2Elements: HTMLHeadingElement[] = Array.from(container.querySelectorAll("h2"))
+
+      h2Elements.forEach((heading, headingIndex) => {
+        const id = `${heading.textContent?.trim().toLowerCase().replace(/\s+/g, "-")}-${containerIndex}-${headingIndex}`
+        heading.setAttribute("id", id)
+        allHeadings.push({text: heading.textContent || "", id})
+      })
+    })
+
+    setHeadings(allHeadings)
+
+    const observerCallback: IntersectionObserverCallback = entries => {
+      entries.forEach(entry => {
+        const top = entry.boundingClientRect.top
+
+        console.log(
+          `Observer | Target: ${entry.target.id} | isIntersecting: ${entry.isIntersecting} | intersectionRatio: ${entry.intersectionRatio}`
+        )
+
+        if (entry.isIntersecting) {
+          setActiveHeading(entry.target.id)
+        }
+      })
+    }
+
+    const observer = new IntersectionObserver(observerCallback, {
+      rootMargin: "0px 0px -80% 0px",
+      threshold: 0,
+    })
+
+    wysiwygContainers.forEach(container => {
+      const h2Elements = container.querySelectorAll("h2")
+      h2Elements.forEach(heading => observer.observe(heading))
+    })
+
+    return () => {
+      observer.disconnect()
+    }
+  }, [])
+
+  const options: SelectOptionDefinition<string>[] = [
+    ...headings.map(heading => ({
+      value: heading.id,
+      label: heading.text,
+    })),
+    {
+      value: "related-links-header",
+      label: relLinkHeading || "Related content",
+      disabled: true,
+    },
+    ...(relLinks?.map(link => ({
+      value: String(link.url),
+      label: String(link.title),
+    })) || []),
+  ]
+
+  return (
+    <div>
+      <div className="block w-full md:w-500 lg:hidden">
+        <SelectList
```

Comment:
I was thinking about this. Instead of using the select list and doing all the logic for it, why not just use an accordion effect on mobile. That will save the logic for the options, the selection handling, and even let you render the links server side.